llvm.sh

```diff
@@ -30,48 +40,32 @@ fi
 # We default to the current stable branch of LLVM
 LLVM_VERSION=$CURRENT_LLVM_STABLE
 ALL=0
-if [ "$#" -ge 1 ]; then
-    LLVM_VERSION=$1
-    if [ "$1" == "all" ]; then
+if [ "$#" -ge 1 ] && [ "${1::1}" != "-" ]; then
+    if [ "$1" != "all" ]; then
+        LLVM_VERSION=$1
+    else
         # special case for ./llvm.sh all
-        LLVM_VERSION=$CURRENT_LLVM_STABLE
         ALL=1
     fi
+    OPTIND=2
     if [ "$#" -ge 2 ]; then
       if [ "$2" == "all" ]; then
           # Install all packages
           ALL=1
+          OPTIND=3
       fi
     fi
 fi
 
+# Set default values for commandline arguments
+BASE_URL="http://apt.llvm.org"
 DISTRO=$(lsb_release -is)
 VERSION=$(lsb_release -sr)
-
-if [[ $EUID -ne 0 ]]; then
-   echo "This script must be run as root!"
-   exit 1
-fi
-
-declare -A LLVM_VERSION_PATTERNS
-LLVM_VERSION_PATTERNS[9]="-9"
-LLVM_VERSION_PATTERNS[10]="-10"
-LLVM_VERSION_PATTERNS[11]="-11"
-LLVM_VERSION_PATTERNS[12]="-12"
-LLVM_VERSION_PATTERNS[13]="-13"
-LLVM_VERSION_PATTERNS[14]="-14"
-LLVM_VERSION_PATTERNS[15]=""
-
-if [ ! ${LLVM_VERSION_PATTERNS[$LLVM_VERSION]+_} ]; then
-    echo "This script does not support LLVM version $LLVM_VERSION"
-    exit 3
-fi
-
-LLVM_VERSION_STRING=${LLVM_VERSION_PATTERNS[$LLVM_VERSION]}
-
-# obtain VERSION_CODENAME and UBUNTU_CODENAME (for Ubuntu and its derivatives)
+UBUNTU_CODENAME=""
+CODENAME_FROM_ARGUMENTS=""
 source /etc/os-release
 DISTRO=${DISTRO,,}
+# obtain VERSION_CODENAME and UBUNTU_CODENAME (for Ubuntu and its derivatives)
```

Comment:
This comment was meant for the `source /etc/os-release` above - thats where you pull these 2 variables in